tests/xharness/Xharness.Tests/Jenkins/ResourceManagerTests.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System.Collections.Generic;
+using System.Resources;
+using Microsoft.DotNet.XHarness.iOS.Shared.Hardware;
+using Xharness.Jenkins;
+using Moq;
+using NUnit.Framework;
+using System.Linq;
+
+namespace Xharness.Tests.Jenkins {
+	[TestFixture]
+	public class ResourceManagerTests {
+
+		Mock<IHardwareDevice> device1;
+		Mock<IHardwareDevice> device2;
+		Xharness.Jenkins.ResourceManager resourceManager;
+		
+		[SetUp]
+		public void SetUp ()
+		{
+			device1 = new Mock<IHardwareDevice> ();
+			device2 = new Mock<IHardwareDevice> ();
+			resourceManager = new Xharness.Jenkins.ResourceManager ();
+		}
+
+		[Test]
+		public void GetDeviceResourceMissingTest ()
+		{
+			var devices = new List<IHardwareDevice> { device1.Object, device2.Object };
+			// set the device expectations
+			device1.Setup (d => d.UDID).Returns ("device1");
+			device1.Setup (d => d.Name).Returns ("device1");
+
+			device2.Setup (d => d.UDID).Returns ("device2");
+			device2.Setup (d => d.Name).Returns ("device2");
+
+			var resources = resourceManager.GetDeviceResources (devices);
+
+			// assert that both are present
+			Assert.NotNull (resources, "resources");
+			Assert.AreEqual (2, resources.Count ());
+			Assert.AreEqual (1, resources.Where (r => r.Name == "device1").Count ());
+			Assert.AreEqual (1, resources.Where (r => r.Name == "device2").Count ());
```

Comment:
you should label each one individually
in case debug symbols (or runtime) is broken and point to the incorrect line (or no line at all)